### PR TITLE
[FIXED JENKINS-48632] Ensure descriptions are included on downstream parameters

### PR DESCRIPTION
**Pull request by abayer**
[JENKINS-48632](https://issues.jenkins-ci.org/browse/JENKINS-48632)

cc @reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -38,10 +38,10 @@
         </pluginRepository>
     </pluginRepositories>
     <properties>
-        <jenkins.version>1.642.3</jenkins.version>
+        <jenkins.version>2.7.3</jenkins.version>
```
Comment (abayer) [inline pom.xml]: Various plugin/core bumps are so we can do `properties(...)` and `parameters: [booleanParam(...)]` and have fairly current (for core 2.7.3, at least) dependencies for tests. I basically lifted all the dependency versions straight from Declarative, which is also on 2.7.3.

File: src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepExecution.java [hunk 2]
```diff
@@ -151,26 +152,27 @@ protected Job asJob() {
...
-                    if (!names.contains(pDef.getName())) {
+                    if (!allParameters.containsKey(pDef.getName())) {
                         ParameterValue defaultP = pDef.getDefaultParameterValue();
                         if (defaultP != null) {
-                            completeListOfParameters.add(defaultP);
+                            allParameters.put(defaultP.getName(), defaultP);
                         }
+                    } else {
+                        // Get the description of specified parameters here. UI submission of parameters uses formatted description.
+                        allParameters.get(pDef.getName()).setDescription(pDef.getFormattedDescription());
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepExecution.java]: That does not accord with what I see [here](https://github.com/jenkinsci/jenkins/blob/90f778d2819cfd223f8815a3d84ce3e904d91fb0/core/src/main/java/hudson/model/StringParameterDefinition.java#L129) (web submission) or [here](https://github.com/jenkinsci/jenkins/blob/90f778d2819cfd223f8815a3d84ce3e904d91fb0/core/src/main/java/hudson/model/StringParameterDefinition.java#L134) (`build` CLI command submission). And I see no test coverage of `description`s that would be transformed by a `MarkupFormatter`.

As of SECURITY-353. [`ParameterValue.description` is not assumed to have been formatted](https://github.com/jenkinsci/jenkins/blob/90f778d2819cfd223f8815a3d84ce3e904d91fb0/core/src/main/resources/hudson/model/StringParameterValue/value.jelly#L30).

File: src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepTest.java [hunk 3]
```diff
@@ -477,4 +479,37 @@ public MultiBranchProjectFactory newInstance() {
...
+    @Test
+    public void parameterDescriptions() throws Exception {
+        WorkflowJob ds = j.jenkins.createProject(WorkflowJob.class, "ds");
+        ds.setDefinition(new CpsFlowDefinition("properties([\n" +
+                "  parameters([\n" +
+                "    booleanParam(defaultValue: true, description: 'flag description', name: 'flag'),\n" +
+                "    string(defaultValue: 'default string', description: 'strParam description', name: 'strParam')\n" +
+                "  ])\n" +
+                "])\n", true));
+        // Define the parameters
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepTest.java]: easier to add the `ParametersDefinitionProperty` programmatically

File: src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepTest.java [hunk 4]
```diff
@@ -477,4 +479,37 @@ public MultiBranchProjectFactory newInstance() {
...
+
+        WorkflowRun r = ds.getBuildByNumber(2);
+        assertNotNull(r);
+
+        ParametersAction action = r.getAction(ParametersAction.class);
+        assertNotNull(action);
+
+        ParameterValue flagValue = action.getParameter("flag");
+        assertNotNull(flagValue);
+        assertEquals("flag description", flagValue.getDescription());
```
Comment (jglick) [inline src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepTest.java]: IIUC this was the one assertion that failed otherwise?

**Comment by jglick** [comment]
> As of SECURITY-353

Perhaps you should bump your baseline to [2.32.2](https://jenkins.io/security/advisory/2017-02-01/) or newer, so that you are testing against what users should actually be running, rather than something which became officially obsolete well over a year ago.

File: src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepExecution.java (hunk 2 above)
Comment (abayer) [reply src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepExecution.java]: Works for me.

File: src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepTest.java (hunk 4 above)
Comment (abayer) [reply src/test/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepTest.java]: Correct.

**Pull request merged**